ladybug/designday.py

```diff
@@ -0,0 +1,1109 @@
+        return "Design Day - {} [{}]".format(
+            self.name, self._day_type)
+
+
+class DryBulbCondition(object):
+    """Represents dry bulb conditions on a design day.
+
+    attributes:
+        max_dry_bulb
+        dry_bulb_range
+        modifier_type
+        modifier_schedule
+    """
+    def __init__(self, max_dry_bulb, dry_bulb_range,
+                 modifier_type='DefaultMultipliers', modifier_schedule=''):
+        """Initalize the class."""
+        self.max_dry_bulb = max_dry_bulb
+        self.dry_bulb_range = dry_bulb_range
+        self.modifier_type = str(modifier_type)
+        self.modifier_schedule = str(modifier_schedule)
+
+    @property
+    def hourly_values(self):
+        """A list of temperature values for each hour over the design day."""
+        return [self._max_dry_bulb - self._dry_bulb_range * x for
+                x in temp_multipliers]
+
+    @property
+    def max_dry_bulb(self):
+        """Get or set the max dry bulb temperature."""
+        return self._max_dry_bulb
+
+    @max_dry_bulb.setter
+    def max_dry_bulb(self, data):
+        assert isinstance(data, (float, int)), 'max_dry_bulb must be an' \
+            ' number. Got {}'.format(type(data))
+        self._max_dry_bulb = data
+
+    @property
+    def dry_bulb_range(self):
+        """Get or set the dry bulb range over the day."""
+        return self._dry_bulb_range
+
+    @dry_bulb_range.setter
+    def dry_bulb_range(self, data):
+        assert isinstance(data, (float, int)), 'dry_bulb_range must be a' \
+            ' number. Got {}'.format(type(data))
+        assert data >= 0, 'dry_bulb_range must be greater than or equal to' \
+            ' zero. Got {}'.format(str(data))
+        self._dry_bulb_range = data
+
+    @property
+    def isDryBulbCondition(self):
+        """Return True."""
+        return True
+
+    def ToString(self):
+        """Overwrite .NET ToString."""
+        return self.__repr__()
+
+    def __repr__(self):
+        """dry bulb condition representation."""
+        return "Dry Bulb Condition [Max: {}, Range: {}]".format(
+            str(self._max_dry_bulb), str(self._dry_bulb_range))
+
+
+class HumidityCondition(object):
+    """Represents humidity conditions on the design day.
+
+    attributes:
+        hum_type: Choose from
+            Wetbulb, Dewpoint, HumidityRatio, Enthalpy
+        hum_value: The value of the condition above
+        barometric_pressure: Default is to use pressure at sea level
+        schedule: Optional humidity schedule
+        wet_bulb_range: Optional wet bulb temperature range
+    """
+    def __init__(self, hum_type, hum_value, barometric_pressure=101325,
+                 schedule='', wet_bulb_range=''):
+        """Initalize the class."""
+        self.hum_type = hum_type
+        self.hum_value = hum_value
+        self.barometric_pressure = barometric_pressure
+        self.schedule = schedule
+        self.wet_bulb_range = wet_bulb_range
```

Comment:
Should it be a list and not a string?  I mean if it is a range.

Reply:
I think you may be missing how design day temperatures work.  You can generate a list of hourly temperatures over the design day from the dry_bulb_max, the range over the day, and the temperature multipliers for each hour (which are standardized).  You can read here about how this works:
https://bigladdersoftware.com/epx/docs/8-9/engineering-reference/climate-calculations.html